### PR TITLE
Enable packaging v6 DBs

### LINKED ISSUE #367: Make grype-db download smaller by switching compression methods
**What would you like to be added**:

Grype should download a smaller file during it's database update, probably by using .zstd compression on the current database schema.

**Why is this needed**:

The Grype database has grown over the years, to the point where now the database is 184 mb as a gzipped tar. This results in load on the CDN, and poor experience for many users.

**Tasks**:

- [x] update yardstick to be able to handle zstd files https://github.com/anchore/yardstick/pull/441
- [x] ~update grype-db-manager config shape to include a suffix per schema version, and pass that suffix as `-e` to `grype-db package`~
- [x] ~update that configuration to pass `-e tar.zstd` for v4 and v5 schemas~ code responds to this, no config update necessary

### PR DESCRIPTION
**Pull request by wagoodman**
#436 covers the `build` subcommand, this PR covers plumbing v6 distribution behavior through the `package` command. The package command supports both v1-5 and v6 behaviors separately. 

The new v6 behavior:
- creates a `.tar.zstd` archive (thus related to https://github.com/anchore/grype-db/issues/367)
- does not embed any `metadata.json` or similar files into the archive

v1-5 behavior is unchanged.

Fixes #367

PR Stack:
1. https://github.com/anchore/grype-db/pull/436

### CODE REVIEW

File: internal/tarutil/writer.go
Comment (willmurphyscode) [inline internal/tarutil/writer.go:77]: Check that the compressor is installed (`zstd` or `xz` is on PATH).

File: internal/tarutil/writer.go
Comment (willmurphyscode) [inline internal/tarutil/writer.go:55]: Please update README to require `zstd` on PATH in setup instructions.

File: pkg/process/package.go
Comment (wagoodman) [inline pkg/process/package.go:112]: pull the latest tests/code from #441

File: pkg/process/package.go
Comment (wagoodman) [inline pkg/process/package.go:122]: should consider skipping `latest.json`

**Review by willmurphyscode — APPROVED** [review]

File: internal/tarutil/writer.go
Comment (wagoodman) [reply internal/tarutil/writer.go:52]: decompression is what is affected here. The `--ultra` option unlocks `-#` values to use when compressing which may use more resources than desirable when attempting to decompress the archive later.

Comment (willmurphyscode) [reply internal/tarutil/writer.go:52]: Does this comment really mean "`--utlra -22` is the highest setting we can get without causing excess memory use in Grype?"

Or does it really mean, "do not pass higher than `--ultra -22` without testing whether Grype starts consuming excess memory?"

It just feels a bit like a vague warning sign right now.

Comment (wagoodman) [reply internal/tarutil/writer.go:52]: 22 is the highest option currently, there are memory impacts at decompression. The comment lists specific memory impacts for two samples of compression levels. This is purely an informational comment (not necessarily a warning) and the final value has yet to be selected while v6 is still in development.

**Comment by wagoodman** [comment]
waiting for PR stack base to get reviewed/merged before attempting to merge this (this should go to main, not the PR stack base branch).

**Comment by wagoodman** [comment]
bottom of the stack merged -- force pushed to rebase on main

**Pull request merged**
